module-system/module-implementations/sov-chain-state/src/lib.rs

```diff
@@ -89,6 +94,17 @@ impl<Cond> TransitionInProgress<Cond> {
     }
 }
 
+/// The information about a transition height
+#[derive(BorshDeserialize, BorshSerialize, Clone, Debug, PartialEq, Eq, PartialOrd, Ord)]
+pub struct TransitionHeight(pub u64);
+
+impl TransitionHeight {
+    /// Returns the inner content of [`TransitionHeight`]
```

Comment:
We're not actually getting any extra invariant safety from `TransitionHeight`. I think we need to change the definition to `pub struct TransitionHeight(u64);` and change the construction logic to make it impossible to create a `TransitionHeight` bigger than the `current_height` stored in the chain state module. 

I think the easiest way to do this is to have a fallible constructor which takes the `max_height` as an argument and fails if the requested height is too large. `pub fn new(requested_height: u64, max_height: u64) -> Result<Self>;`

Reply:
We managed to get around that extra new type by adding an explicit check of the invariant.